From f8beaa14bbc4e2b0a5a9b48e1d8b0efb5f05f5b2 Mon Sep 17 00:00:00 2001
From: Chad Dupuis <cdupuis@redhat.com>
Date: Mon, 18 Aug 2014 19:56:44 -0400
Subject: [scsi] qla2xxx: Use proper log message for flash lock failed error

Message-id: <1408391817-6389-57-git-send-email-cdupuis@redhat.com>
Patchwork-id: 87951
O-Subject: [PATCH RHEL 7.1 e-stor 56/69] qla2xxx: Use proper log message for flash lock failed error.
Bugzilla: 1089346
RH-Acked-by: Rob Evers <revers@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Atul Deshmukh <atul.deshmukh@qlogic.com>

scsi commit id 27f4b72f94208f9f82589214d5dfdae57d74f580

Signed-off-by: Atul Deshmukh <atul.deshmukh@qlogic.com>
Signed-off-by: Saurav Kashyap <saurav.kashyap@qlogic.com>
Signed-off-by: Christoph Hellwig <hch@lst.de>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/qla2xxx/qla_dbg.c b/drivers/scsi/qla2xxx/qla_dbg.c
index 16349b7..75d41f2 100644
--- a/drivers/scsi/qla2xxx/qla_dbg.c
+++ b/drivers/scsi/qla2xxx/qla_dbg.c
@@ -50,7 +50,7 @@
  * |                              |                    | 0x800b,0x8039  |
  * | AER/EEH                      |       0x9011       |  |
  * | Virtual Port                 |       0xa007       |  |
- * | ISP82XX Specific             |       0xb155       | 0xb002,0xb024  |
+ * | ISP82XX Specific             |       0xb157       | 0xb002,0xb024  |
  * |                              |                    | 0xb09e,0xb0ae  |
  * |                              |                    | 0xb0e0-0xb0ef  |
  * |                              |                    | 0xb085,0xb0dc  |
diff --git a/drivers/scsi/qla2xxx/qla_nx.c b/drivers/scsi/qla2xxx/qla_nx.c
index 3d77bf7..2801aee 100644
--- a/drivers/scsi/qla2xxx/qla_nx.c
+++ b/drivers/scsi/qla2xxx/qla_nx.c
@@ -848,6 +848,7 @@ qla82xx_rom_lock(struct qla_hw_data *ha)
 {
  int done = 0, timeout = 0;
  uint32_t lock_owner = 0;
+ scsi_qla_host_t *vha = pci_get_drvdata(ha->pdev);
 
  while (!done) {
   /* acquire semaphore2 from PCI HW block */
@@ -856,6 +857,9 @@ qla82xx_rom_lock(struct qla_hw_data *ha)
    break;
   if (timeout >= qla82xx_rom_lock_timeout) {
    lock_owner = qla82xx_rd_32(ha, QLA82XX_ROM_LOCK_ID);
+   ql_log(ql_log_warn, vha, 0xb157,
+       "%s: Simultaneous flash access by following ports, active port = %d: accessing port = %d",
+       __func__, ha->portnum, lock_owner);
    return -1;
   }
   timeout++;
diff --git a/drivers/scsi/qla2xxx/qla_nx2.c b/drivers/scsi/qla2xxx/qla_nx2.c
index 3c39446..a8e2603 100644
--- a/drivers/scsi/qla2xxx/qla_nx2.c
+++ b/drivers/scsi/qla2xxx/qla_nx2.c
@@ -448,8 +448,8 @@ qla8044_flash_lock(scsi_qla_host_t *vha)
    lock_owner = qla8044_rd_reg(ha,
        QLA8044_FLASH_LOCK_ID);
    ql_log(ql_log_warn, vha, 0xb113,
-       "%s: flash lock by %d failed, held by %d\n",
-    __func__, ha->portnum, lock_owner);
+       "%s: Simultaneous flash access by following ports, active port = %d: accessing port = %d",
+       __func__, ha->portnum, lock_owner);
    ret_val = QLA_FUNCTION_FAILED;
    break;
   }
-- 
1.7.1